src/MongoDB.Driver.Core/Core/NativeLibraryLoader/ILibraryLocator.cs

```diff
@@ -0,0 +1,61 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.IO;
+using System.Reflection;
+
+namespace MongoDB.Driver.Core.NativeLibraryLoader
+{
+    internal interface ILibraryLocator
+    {
+        string GetLibraryAbsolutePath(SupportedPlatform currentPlatform);
+    }
+
+    internal abstract class LibraryRelativeLocatorBase : ILibraryLocator
+    {
+        public virtual Assembly Assembly => typeof(LibraryRelativeLocatorBase).GetTypeInfo().Assembly;
+
+        public abstract string GetLibraryRelativePath(SupportedPlatform currentPlatform);
+
+        public string GetLibraryAbsolutePath(SupportedPlatform currentPlatform)
+        {
+            var relativePath = GetLibraryRelativePath(currentPlatform);
+            return GetAbsolutePath(relativePath);
+        }
+
+        // private methods
+        private string FindLibraryOrThrow(string basePath, string relativePath)
+        {
+            var fullPath = Path.Combine(basePath, relativePath);
```

Comment:
Perhaps name varilable `absolutePath` for consistency?

Reply:
Consider refactoring to:

    private string FindLibraryOrThrow(string basePath, string relativePath)
    {
        var absolutePath = Path.Combine(basePath, relativePath);
        if (!File.Exists(absolutePath))
        {
            throw new FileNotFoundException($"Could not find library {absolutePath}.");
        }

        return absolutePath;
    }

It's a minor nit: most commonly we check error conditions first and then return a value at the end.

But you can disregard this suggestion if you don't like it.

Reply:
done